src/mina-signer/src/rosetta.ts

```diff
@@ -124,24 +241,53 @@ function rosettaTransactionToSignedCommand({
   };
 }
 
-type RosettaTransactionJson = {
+type UnsignedPayload = {
+  unsigned_transaction: UnsignedTransaction;
+  payloads: any[];
+}
+
+type UnsignedTransaction = {
+  randomOracleInput: string;
+  signerInput: {
+    prefix: string[];
+    suffix: string[];
+  };
+  payment: Payment | null;
+  stakeDelegation: StakeDelegation | null;
+}
+
+type SignedTransaction = {
   signature: string;
-  payment: {
-    to: string;
-    from: string;
-    fee: string;
-    token: string;
-    nonce: string;
-    memo: string | null;
-    amount: string;
-    valid_until: string | null;
-  } | null;
-  stake_delegation: {
-    delegator: string;
-    new_delegate: string;
-    fee: string;
-    nonce: string;
-    memo: string | null;
-    valid_until: string | null;
-  } | null;
-};
+  payment: Payment | null;
+  stake_delegation: StakeDelegation | null;
+}
+
+type RosettaSignature = {
+  hex_bytes: string;
+  public_key: {
+    hex_bytes: string;
+    curve_type: string;
+  };
+  signature_type: string;
+  signing_payload: any;
```

Comment:
can we change all the `any` to `unknown` for type safety?
```suggestion
  signing_payload: unknown;
```